health-checker/src/main/java/io/stargate/health/StargateNodesHealthChecker.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.health;
+
+import com.codahale.metrics.health.HealthCheck;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.db.datastore.DataStoreFactory;
+import io.stargate.db.datastore.ResultSet;
+import io.stargate.db.datastore.Row;
+import java.util.UUID;
+import java.util.concurrent.Future;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StargateNodesHealthChecker extends HealthCheck {
```

Comment:
`system.local` and `system.peers` are relevant to CQL drivers. IIRC, we had some cases when those tables disappeared due to `nodetool resetlocalschema`.

Reply:
If we decide to keep it, I'd prefer to add `system.peers` to `DataStoreHealthChecker` and put storage-level queries in a new class to reduce code disturbance since `DataStoreHealthChecker` already deals with `system.local`.

Reply:
@dimas-b 
`"If we decide to keep it, I'd prefer to add system.peers to DataStoreHealthChecker and put storage-level queries in a new class to reduce code disturbance since DataStoreHealthChecker already deals with system.local."`
I've already done it this way, isn't it?

Reply:
Ah... might be GH missing the rename... the diff looks like a big change in the old `DataStoreHealthChecker` plus old code showing as "new" in `StargateNodesHealthChecker` :shrug: